From bd476b3055b1fe57bb81dd78dddaf4aa2a0cd9d7 Mon Sep 17 00:00:00 2001
From: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Date: Thu, 5 Mar 2015 00:07:33 +0100
Subject: [PATCH] Use correct uname identifiers in arch_map for SuperH
 architecture

https://bugs.freedesktop.org/show_bug.cgi?id=89421
(cherry picked from commit 3a867c6a2361c8af943d3ed452da6e8623a3f65d)
(cherry picked from commit 9b6d1109707911b63b76b1c7a26e89a21cef9011)
---
 src/shared/architecture.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/shared/architecture.c b/src/shared/architecture.c
index dc45f3589d..617f39cfc6 100644
--- a/src/shared/architecture.c
+++ b/src/shared/architecture.c
@@ -108,8 +108,12 @@ Architecture uname_architecture(void) {
                 { "armv8l",     ARCHITECTURE_ARM      },
                 { "armv8b",     ARCHITECTURE_ARM_BE   },
 #elif defined(__sh__) || defined(__sh64__)
-                { "sh64",       ARCHITECTURE_SH64     },
-                { "sh",         ARCHITECTURE_SH       },
+                { "sh5",        ARCHITECTURE_SH64     },
+                { "sh2",        ARCHITECTURE_SH       },
+                { "sh2a",       ARCHITECTURE_SH       },
+                { "sh3",        ARCHITECTURE_SH       },
+                { "sh4",        ARCHITECTURE_SH       },
+                { "sh4a",       ARCHITECTURE_SH       },
 #elif defined(__m68k__)
                 { "m68k",       ARCHITECTURE_M68K     },
 #elif defined(__tilegx__)
